script/validate.go

```diff
@@ -0,0 +1,167 @@
+package script
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/halseth/tapsim/output"
+	"github.com/pkg/term"
+)
+
+// Validate builds a tap leaf using the passed pkScript and executes it step by
+// step with the provided witness.
+func Validate(tx *wire.MsgTx,
```

Comment:
This shouldn't really be its own function, that is mostly a copy of the Execute function. We should rather aim for extracting the common parts into a function that executes the fully built tx, such that the logic can be shared.

Here's a rough draft of a starting point: https://github.com/halseth/tapsim/commit/709f8710480b650bcae322333a49ac2a68f6fa0b

Reply:
Yeah I wanted to put it in execute but didn't really know what i was doing so found it easier to make my own function, Your starting point helps a lot, will move to that

Reply:
have draft pr here #9, however after going through idk if it is the best approach, gets really messy keeping the generation stuff with passing in everything, feels a lot cleaner as a separate function

Reply:
Sorry for taking some time with the feedback, but I have been pondering how to solve this as clean as possible.

Could something like this work? https://github.com/halseth/tapsim/commits/tx-validate/

Check if it works for you. I haven't tested the code, so you might have to alter it for your use case.